common/lib/xmodule/xmodule/js/src/video/00_sjson.js

```diff
@@ -51,10 +61,44 @@ function() {
             return min;
         };
 
+        function filter(start, end) {
+            var filteredTimes = [];
+            var filteredCaptions = [];
+            var startTimes = getStartTimes();
+            var captions = getCaptions();
+            var currentStartTime;
+            var i;
+
+            if (startTimes.length !== captions.length) {
```

Comment:
I'm not sure, but I think you're checking for valid inputs here. If these are not the same, are we in an invalid state? And if so, don't we want to raise an error rather than just returning an empty array?